### PR TITLE
Removing "functional" from "functional-reactive" term.

### PR DESCRIPTION
**Pull request by Sfshaza**
Staged: https://sz-flutter.firebaseapp.com/faq/

### CODE REVIEW

**Comment by Hixie** [comment]
I'm assuming that is the wrong bug link, and that you meant to link to https://github.com/flutter/flutter/issues/11474.

File: faq.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ built with Flutter to users. We'd love to hear what you're building!
 ### What is inside the Flutter SDK?
 
 * Heavily optimized, mobile-first 2D rendering engine with excellent support for text
-* Functional-reactive framework
+* Reactive framework
```
Comment (Hixie) [inline faq.md]: We're no more reactive than functional, so I disagree that this is a correct change.

Comment (wmleler) [inline faq.md]: Modern, reactive-style framework

File: faq.md [hunk 2]
```diff
@@ -129,9 +129,8 @@ and one of the many editors that support [editing Dart](https://www.dartlang.org
 
 ### Does Flutter come with a framework?
 
-Yes! Flutter ships with a functional-reactive style framework, inspired by
-React. Flutter's framework is designed to be layered and customizable
-(and optional).
+Yes! Flutter ships with a reactive style framework, inspired by React.
```
Comment (wmleler) [inline faq.md]: modern reactive-style

File: faq.md (hunk 1 above)
Comment (Hixie) [inline faq.md]: I think it'd be fine to remove the label "reactive" as well and just say something like "Modern UI framework"

File: faq.md (hunk 2 above)
Comment (Hixie) [inline faq.md]: How about "Flutter ships with a modern framework, inspired by React"?

File: technical-overview.md [hunk 3]
```diff
@@ -35,19 +35,20 @@ What are some advantages of Flutter? It helps you:
...
-of OEM widget sets
+    *   Benefit from a rich set of Material Design and Cupertino (iOS-flavor)
+        widgets built using Flutter's own framework
+    *   Realize custom, beautiful, brand-driven designs, without the
+        limitations of OEM widget sets
 
 ## Core principles
 
-Flutter includes a functional-reactive framework, a 2D rendering engine, ready-made
+Flutter includes a reactive framework, a 2D rendering engine, ready-made
```
Comment (wmleler) [inline technical-overview.md]: modern reactive-style

File: widgets-intro.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ permalink: /widgets-intro/
 
 ## Introduction
 
-Flutter widgets are built using a functional-reactive framework, which takes
+Flutter widgets are built using a reactive framework, which takes
```
Comment (wmleler) [inline widgets-intro.md]: modern reactive-style

File: faq.md [hunk 5]
```diff
@@ -233,10 +232,10 @@ The primary criteria we used to pick a programming language were the following:
   performance and delivers predictable performance, without periodic
   pauses that would cause dropped frames.
 
-* _Fast allocation_. The Flutter framework uses a functional-reactive style
+* _Fast allocation_. The Flutter framework uses a reactive style
```
Comment (Hixie) [inline faq.md]: It's actually the functional part that's more relevant here, not the reactive part. The reactive programming model doesn't necessarily imply much memory allocation. What's relevant here is that we use a functional-style flow where many immutable objects get allocated to describe the scene, then the previous scene's objects get released.

File: faq.md [hunk 6]
```diff
@@ -233,10 +232,10 @@ The primary criteria we used to pick a programming language were the following:
...
   programming model, whose performance depends heavily on the underlying memory
   allocator efficiently handling small, short-lived allocations. The
-  functional-reactive style was developed in languages with this property and
+  reactive style was developed in languages with this property and
```
Comment (Hixie) [inline faq.md]: Is this statement still accurate?

The "functional-reactive" style was developed in different languages than the "reactive" style. I would be wary of changing this statement without verifying the accuracy of the new claim.

**Comment by Hixie** [comment]
Our framework is not "reactive-style". We don't use streams at all in the framework itself, for instance. The framework's pipeline is not asynchronous. The primary data flow doesn't use events.

File: faq.md [hunk 7]
```diff
@@ -233,11 +233,9 @@ The primary criteria we used to pick a programming language were the following:
...
   pauses that would cause dropped frames.
 
-* _Fast allocation_. The Flutter framework uses a functional-reactive style
-  programming model, whose performance depends heavily on the underlying memory
-  allocator efficiently handling small, short-lived allocations. The
-  functional-reactive style was developed in languages with this property and
-  does not work efficiently in languages that lack this facility.
+* _Fast allocation_. The Flutter framework uses a functional-style flow
+  that depends heavily on the underlying memory
+  allocator efficiently handling small, short-lived allocations.
```
Comment (Hixie) [inline faq.md]: I'd like to keep the second sentence if we can. It makes it clearer why this is something Dart offers but that, say, Java does not.

**Comment by Hixie** [comment]
The update is better, but I think we should really avoid the term "reactive-style". It's even more inaccurate than "functional reactive", IMHO.

What we really need is a term for the specific paradigm that we actually most prominently use, the whole build function thing, which I think is what most people are getting at with "functional-reactive" when they use it loosely. I really can't find a term that is used to describe this. None of the React marketing materials seem to talk about the style of programming used in the framework.

File: faq.md (hunk 7 above)
Comment (Sfshaza) [inline faq.md]: @Hixie , who could validate that second sentence? Kevmoo? Someone else on the Dart team?

Comment (Hixie) [inline faq.md]: Well if we just leave it the way it was then I don't think we need to particularly change anything. :-)

@abarth is the one who originally wrote this sentence so he might have thoughts on how to rephrase it if we want to remove mention of "functional-reactive" but keep it accurate.

Comment (abarth) [inline faq.md]: You could just say "This style was developed ..."

File: faq.md [hunk 8]
```diff
@@ -110,8 +110,9 @@ built with Flutter to users. We'd love to hear what you're building!
 
 ### What is inside the Flutter SDK?
 
-* Heavily optimized, mobile-first 2D rendering engine with excellent support for text
-* Functional-reactive framework
+* Heavily optimized, mobile-first 2D rendering engine with excellent support for
+  text
+* Modern reactive framework
```
Comment (Hixie) [inline faq.md]: How about "Modern react-style framework"?

Though personally I still think we should say "functional-reactive". I don't really understand what's wrong with the term.

File: faq.md (hunk 7 above)
Comment (Hixie) [inline faq.md]: I think the original text is fine, but if we have to change it, I would recommend something like:

"_Fast allocation_. The Flutter framework uses a react-style programming model, whose performance depends heavily on the underlying memory allocator efficiently handling small, short-lived allocations. This style was developed in languages with this property and does not work efficiently in languages that lack this facility."

File: technical-overview.md [hunk 9]
```diff
@@ -35,21 +35,22 @@ What are some advantages of Flutter? It helps you:
...
+        widgets built using Flutter's own framework
+    *   Realize custom, beautiful, brand-driven designs, without the
+        limitations of OEM widget sets
 
 ## Core principles
 
-Flutter includes a functional-reactive framework, a 2D rendering engine, ready-made
-widgets, and development tools. These components work together to help you
-design, build, test, and debug apps. Everything is organized around a few core
+Flutter includes a modern modern reactive framework, a 2D rendering engine,
```
Comment (Hixie) [inline technical-overview.md]: replace "modern modern reactive" with "react-style" or "functional-reactive". (I'd prefer the second.)

File: widgets-intro.md [hunk 10]
```diff
@@ -9,7 +9,7 @@ permalink: /widgets-intro/
 
 ## Introduction
 
-Flutter widgets are built using a functional-reactive framework, which takes
+Flutter widgets are built using a modern reactive framework, which takes
```
Comment (Hixie) [inline widgets-intro.md]: Similarly, "modern react-style framework" or "functional-reactive framework".

**Comment by Hixie** [comment]
I'm less and less convinced that this change is an improvement.

File: faq.md (hunk 7 above)
Comment (Sfshaza) [inline faq.md]: OK, I will change it to this. Though I ran the original text past @a-siva and he suggested removing it.

**Comment by Hixie** [comment]
I would prefer not making the change at all, but I can live with this version.
I'll let someone who actually wants the change review it.

**Comment by Sfshaza** [comment]
Wm just lgtm'd it again (after recent changes). Thanks for all feedback.

**Pull request merged**